### PR TITLE
Fix bug unhandle maxium inner transaction 

### PR DESCRIPTION
**Pull request by vuvoth**
## Proposed Changes

- Only handle 256 transaction in one call. 
- Fix bug and add test script.

### CODE REVIEW

File: packages/runtime/src/errors/errors-list.ts [hunk 1]
```diff
@@ -544,6 +544,12 @@ const runtimeGeneralErrors = {
 		title: "%appName% already used",
 		description: "%appName% already used. Please choose another name.",
 	},
+	TOO_MANY_INNER_TXN: {
+		number: 1320,
+		message: "Too many inner transaction",
```
Comment (rmeena840) [inline packages/runtime/src/errors/errors-list.ts:549]:
```suggestion
		message: "Too many inner transaction.",
```

File: packages/runtime/test/fixtures/group-inner-transaction/assets/limit-number-txn.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    )
+
+    program = Cond(
+        [Txn.application_id() == Int(0), on_initialize],
+        [Txn.application_args[0] == Bytes("exec"), on_payment_action]
+    )
+
+    return program
+
+
```
Comment (rmeena840) [inline packages/runtime/test/fixtures/group-inner-transaction/assets/limit-number-txn.py:130]:
```suggestion
```

File: packages/runtime/test/integration/group-inner-transaction.ts [hunk 3]
```diff
@@ -74,4 +76,89 @@ describe("Group inner transaction", function () {
...
+
+	describe("Number inner transaction limitation", function () {
+		let secondAppInfo: AppInfo;
+		this.beforeEach(() => {
+			secondAppInfo = runtime.deployApp(
+				alice.account,
+				{
+					appName: "secondApp",
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: "limit-number-txn.py",
```
Comment (rmeena840) [inline packages/runtime/test/integration/group-inner-transaction.ts:88]: For approval, we are passing pyteal but for the clear program, we are passing teal file. Is this expected?

File: packages/runtime/test/integration/group-inner-transaction.ts [hunk 4]
```diff
@@ -74,4 +76,89 @@ describe("Group inner transaction", function () {
...
+				payFlags: {
+					totalFee: 1000,
+				},
+			};
+
+			runtime.executeTx([txnParams]);
+			syncAccounts();
+		});
+
+		it("Should fail when issue more than 256 inner txn", () => {
```
Comment (rmeena840) [inline packages/runtime/test/integration/group-inner-transaction.ts:113]: @vuvoth Will this fail, here are passing only 4 txn?

**Review by rmeena840 — CHANGES_REQUESTED** [review]
Reviewed.

File: packages/runtime/src/ctx.ts [hunk 5]
```diff
@@ -54,6 +54,7 @@ export class Ctx implements Context {
 	// inner transaction props
 	isInnerTx: boolean; // true if "ctx" is switched to an inner transaction
 	innerTxAppIDCallStack: number[];
+	remainingTxn: number;
```
Comment (ratik21) [inline packages/runtime/src/ctx.ts:57]:
```suggestion
	remainingTxns: number;
```

File: packages/runtime/test/fixtures/group-inner-transaction/assets/limit-number-txn.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            TxnField.type_enum: TxnType.Payment,
+            TxnField.receiver: Txn.sender(),
+            TxnField.amount: Int(0)
+        }),
+        InnerTxnBuilder.Next(),
+        InnerTxnBuilder.SetFields({
+            TxnField.type_enum: TxnType.Payment,
+            TxnField.receiver: Txn.sender(),
+            TxnField.amount: Int(0)
+        }),
```
Comment (ratik21) [inline packages/runtime/test/fixtures/group-inner-transaction/assets/limit-number-txn.py:78]: TEAL supports looping. Can we push inner transactions using a loop?

File: packages/runtime/test/integration/group-inner-transaction.ts [hunk 7]
```diff
@@ -74,4 +76,89 @@ describe("Group inner transaction", function () {
 		// spend 5000 micro Algo
 		assert.equal(contractBalance, contractAcc.balance() + 5000n);
 	});
+
+	describe("Number inner transaction limitation", function () {
```
Comment (ratik21) [inline packages/runtime/test/integration/group-inner-transaction.ts:80]:
```suggestion
	describe("inner transaction limit tealv6", function () {
```

File: packages/runtime/test/integration/group-inner-transaction.ts [hunk 8]
```diff
@@ -74,4 +76,89 @@ describe("Group inner transaction", function () {
...
+				type: types.TransactionType.CallApp,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				appID: secondAppInfo.appID,
+				appArgs: ["str:exec"],
+				payFlags: {
+					totalFee: 1000,
+					note: "second",
+				},
+			};
```
Comment (ratik21) [inline packages/runtime/test/integration/group-inner-transaction.ts:134]: We should use a function for this. In general if you're repeating any code more than 3 times, you should avoid it. You can create a function

File: packages/runtime/test/integration/group-inner-transaction.ts (hunk 4 above)
Comment (vuvoth) [reply packages/runtime/test/integration/group-inner-transaction.ts:113]: No this is more than 256 inner transaction. You can see one inner payment is one inner and we issue 16 * 4 for one single transaction => total transaction is (4 + 16 * 4 * 4) > 256 ( not include big one)

File: packages/runtime/test/integration/group-inner-transaction.ts (hunk 8 above)
Comment (vuvoth) [reply packages/runtime/test/integration/group-inner-transaction.ts:134]: Yeah, good idea!

File: packages/runtime/test/fixtures/group-inner-transaction/assets/limit-number-txn.py [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+        InnerTxnBuilder.Next(),
+        InnerTxnBuilder.SetFields({
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply packages/runtime/test/fixtures/group-inner-transaction/assets/limit-number-txn.py:97]: good idea, let's do that!

**Review by rmeena840 — APPROVED** [review]

**Pull request merged**